tests/pipelines/visualcloze/test_pipeline_visualcloze.py

```diff
@@ -0,0 +1,260 @@
+            "upsampling_height": 32,
+            "upsampling_width": 32,
+            "max_sequence_length": 48,
+            "output_type": "np",
+            "upsampling_strength": 0.4,
+        }
+        return inputs
+
+    def test_visualcloze_different_prompts(self):
+        pipe = self.pipeline_class(**self.get_dummy_components()).to(torch_device)
+
+        inputs = self.get_dummy_inputs(torch_device)
+        output_same_prompt = pipe(**inputs).images[0]
+
+        inputs = self.get_dummy_inputs(torch_device)
+        inputs["content_prompt"] = "A different landscape with forests and rivers"
+        output_different_prompts = pipe(**inputs).images[0]
+
+        max_diff = np.abs(output_same_prompt - output_different_prompts).max()
+
+        # Outputs should be different
+        assert max_diff > 1e-6
+
+    def test_visualcloze_image_output_shape(self):
+        pipe = self.pipeline_class(**self.get_dummy_components()).to(torch_device)
+        inputs = self.get_dummy_inputs(torch_device)
+
+        height_width_pairs = [(32, 32), (72, 57)]
+        for height, width in height_width_pairs:
+            expected_height = height - height % (pipe.vae_scale_factor * 2)
+            expected_width = width - width % (pipe.vae_scale_factor * 2)
+
+            inputs.update({"upsampling_height": height, "upsampling_width": width})
+            image = pipe(**inputs).images[0][0]
+            output_height, output_width, _ = image.shape
+            assert (output_height, output_width) == (expected_height, expected_width)
+
+    def test_inference_batch_single_identical(self):
+        self._test_inference_batch_single_identical(expected_max_diff=1e-3)
+
+    def test_upsampling_strength(self, expected_min_diff=1e-1):
+        pipe = self.pipeline_class(**self.get_dummy_components()).to(torch_device)
+        inputs = self.get_dummy_inputs(torch_device)
+
+        # Test different upsampling strengths
+        inputs["upsampling_strength"] = 0.2
+        output_no_upsampling = pipe(**inputs).images[0]
+
+        inputs["upsampling_strength"] = 0.8
+        output_full_upsampling = pipe(**inputs).images[0]
+
+        # Different upsampling strengths should produce different outputs
+        max_diff = np.abs(output_no_upsampling - output_full_upsampling).max()
+        assert max_diff > expected_min_diff
+
+    def test_different_task_prompts(self, expected_min_diff=1e-1):
+        pipe = self.pipeline_class(**self.get_dummy_components()).to(torch_device)
+        inputs = self.get_dummy_inputs(torch_device)
+
+        output_original = pipe(**inputs).images[0]
+
+        inputs["task_prompt"] = "A different task description for image generation"
+        output_different_task = pipe(**inputs).images[0]
+
+        # Different task prompts should produce different outputs
+        max_diff = np.abs(output_original - output_different_task).max()
+        assert max_diff > expected_min_diff
+
+    def _test_inference_batch_single_identical(
```

Comment:
Why do we have to override it?

Reply:
Hello, we have modified the way the pipe returns images to avoid override.